modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaUtils.java

```diff
@@ -45,38 +45,53 @@ public static SchemaDescriptor prepareSchemaDescriptor(int schemaVer, TableView
      * Prepares column mapper.
      *
      * @param oldDesc Old schema descriptor.
-     * @param oldTblColumns Old columns configuration.
      * @param newDesc New schema descriptor.
-     * @param newTblColumns New columns configuration.
      * @return Column mapper.
      */
     public static ColumnMapper columnMapper(
             SchemaDescriptor oldDesc,
-            NamedListView<? extends ColumnView> oldTblColumns,
-            SchemaDescriptor newDesc,
-            NamedListView<? extends ColumnView> newTblColumns
+            SchemaDescriptor newDesc
     ) {
-        ColumnMapper mapper = null;
+        // Rename not supported from standard, thus only deletion.
+        if (newDesc.keyColumns().length() != oldDesc.keyColumns().length()) {
+            Optional<Column> droppedKeyCol = Arrays.stream(oldDesc.keyColumns().columns())
+                    .filter(c -> !newDesc.column(c.schemaIndex()).name().equals(c.name()))
+                    .findAny();
+
```

Comment:
I barely know anything related to schema logic, but the old algorithm of selecting `droppedKeyCol` seems more solid to me, are you sure the new one is better? What guarantees that the key columns in new schema will have the same indexes as corresponding ones in old schema?

Reply:
I suppose that it absolutely correct approach, plz check how this schemaIndex`es are creating : org.apache.ignite.internal.schema.Columns#sortedCopy thus if columns have different indexes it`s occasion to rise a red flag )